core/services/run_manager.go

```diff
@@ -21,20 +21,23 @@ import (
 )
 
 var (
-	numberRunsExecuted = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_started",
-		Help: "The total number of runs that have run",
-	})
-	numberRunsResumed = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_resumed",
-		Help: "The total number of run resumptions",
-	})
-	numberRunsCancelled = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_cancelled",
-		Help: "The total number of run cancellations",
-	})
+	promTotalRunUpdates = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "run_manager_status_update_total",
+		Help: "The total number of status updates for Job Runs",
+	},
+		[]string{"job_spec_id", "from_status", "status"},
+	)
 )
 
+func promInstrumentJobRunUpdate(from models.RunStatus) func(**models.ID, *models.RunStatus) {
```

Comment:
Couldn't this be `models.ID, models.RunStatus` ? Why double and single indirection here? These things are small so can be copied and are not modified, so don't need to be pointers.

Reply:
The function is created to be deferred, hence the references.  The updates are counted after the calls has finished using initial and final values.  Wish I could point to an article, but this is from past coding practices.

Reply:
Can you find an example article? This doesn't make any sense to me.

Reply:
Tried the change: https://github.com/smartcontractkit/chainlink/commit/1421b77b7a588e465cef24623dd7bc2da4a17008. Passes CI.

If there is some reason to do this, then it'd be nice to have a test that proves it.

Reply:
Unfortunately there’s I couldn’t find an easy way to cover our Prometheus metrics testing without adding an interface.  I suggest we look into this as a separate topic.

Reply:
I don't see the need or the usage of an interface here?

Reply:
You're right, but it was more about whether test cover would be nice for prometheus metrics